File: 240212/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.py
m = int(input())
a, b = tuple(map(int, input().split()))

min_num = 21e8
max_num = 0

def calc(num):
    left = 1
    right = m
    cnt = 0

    while left <= right:
        cnt += 1
        mid = (left + right) // 2

        if mid == num:
            break

        if mid > num:
            right = mid - 1
        else:
            left = mid + 1

    return cnt

for i in range(a, b + 1):
    min_num = min(min_num, calc(i))
    max_num = max(max_num, calc(i))

print(min_num, max_num)

File: 240212/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.cpp
#include <iostream>
#include <algorithm>

using namespace std;

long long m, a, b;
long long min_num = 21e8;
long long max_num = 0;

long long Calc(int num) {
	long long left = 1;
	long long right = m;
	long long cnt = 0;

	while (left <= right) {
		cnt++;
		long long mid = (left + right) / 2;

		if (mid == num) {
			break;
		}

		if (mid > num) {
			right = mid - 1;
		}
		else {
			left = mid + 1;
		}
	}

	return cnt;
}

int main() {
	cin >> m >> a >> b;

	for (int i = a; i <= b; i++) {
		min_num = min(min_num, Calc(i));
		max_num = max(max_num, Calc(i));
	}

	cout << min_num << " " << max_num;
}